python_modules/dagster/dagster_tests/asset_defs_tests/test_assets.py

```diff
@@ -1244,17 +1280,20 @@ def op_1(context):
         if "out_2" in context.selected_output_names:
             yield Output(1, output_name="out_2")
 
-    @op
-    def op_2(x):
-        return x
+    @op(ins={"x": In(Nothing)})
```

Comment:
Ok so the reason the `Nothing`-type input here is now necessary is that the new logic is making it so that assets are now (correctly) being materialized without their parent assets, and this materialization is happening against an empty instance (so otherwise you'd get an error on input load), is that right?

Reply:
Precisely precisely